### PR TITLE
refactor/add_deprecated_module

### PR DESCRIPTION
**Pull request by NeonJarbas**
move all dead/backwards compat only code into a dedicated module  - replaces #89 

deprecated modules are:
- mycroft.audio.services (moved to OPM)
- mycroft.client
  - mycroft.client.enclosure (replaced by PHAL)
  - mycroft.client.speech (moved to mycroft.listener)
  - mycroft.client.text (moved to [OpenVoiceOS/ovos_cli_client](https://github.com/OpenVoiceOS/ovos_cli_client) )
- mycroft.skills.msm_wrapper
- mycroft.tts (moved to mycroft.audio + OPM)
- mycroft.stt (moved to mycroft.listener + OPM)

some other misc deprecated helper classes have also been moved to the new module:
- DevicePrimer (skills)
- NoiseTracker (speech)
- WordExtractor (speech)
- RollingMean (speech)

compatibility is broken for TTS/STT, missing compat modules and unittests should be added in a future PR

### CODE REVIEW

File: mycroft/audio/listener/__init__.py [hunk 1]
```diff
@@ -0,0 +1,449 @@
+# Copyright 2017 Mycroft AI Inc.
```
Comment (NeonDaniel) [inline mycroft/audio/listener/__init__.py:1]: Why is this moved from the speech module to audio?

File: mycroft/deprecated/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"""
+This module contains dead code only
+All code here is in life support and available only for backwards api compatibility
+Bug fixes will be made to maintain compatibility but are very low priority!
+"""
```
Comment (NeonDaniel) [inline mycroft/deprecated/__init__.py:5]: Consider logging a deprecation warning here to cover any deprecated imports

File: mycroft/audio/stt.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+from mycroft.api import STTApi, HTTPError
```
Comment (NeonDaniel) [inline mycroft/audio/stt.py:1]: Why is this moved from speech module?

**Review by NeonDaniel — COMMENTED** [review]
Why is the listener module in audio now? The current structure that separates the code by thread (speech/audio) I think makes more sense

File: mycroft/audio/stt.py (hunk 3 above)
Comment (JarbasAl) [reply mycroft/audio/stt.py:1]: the aim is to deprecate the whole "client" folder here

the decision to move it into audio came from a discussion with @forslund some time ago in mattermost, i believe he intended to do something similar

@forslund pinging you because your feedback is appreciated about this change

File: mycroft/deprecated/__init__.py (hunk 2 above)
Comment (JarbasAl) [reply mycroft/deprecated/__init__.py:5]: deprecation warnings are already in relevant places, this is only documentation for users looking at the code

logging here would be super spammy since it may be imported in some places for backward compat

**Comment by JarbasAl** [comment]
I even considered making speech/audio a single service, but that seemed like a bigger change and breaking.... 

is there any circumstance where we want to run only audio or only speech? do we not always want both?

They very much seem like 2 sides of the same coin to me, but also cleaner if they are separated processes. Hence moving them all to same module but keeping different services

**Comment by forslund** [comment]
Hey,

This is a bit different from my thoughts but I think it's an improvement upon my idea. For me the important thing was that the (messagebus) Origin of the query (STT) and the destination for output of the answer (TTS) the same.

Since the invention of the plugin system the different client processes doesn't make as much sense as they did so moving the core parts for speech and things to a "library" sounds like a good plan. I'm not very well versed in the OVOS architecture but to me this restructuring makes sense.

Also speech -> listener: such a good naming change!

**Review by NeonDaniel — APPROVED** [review]
Looks good; didn't test
